docs/EXTENDED-ARCHIVES.md

```diff
@@ -0,0 +1,344 @@
+
+Declaring compatibility allows images that target a specific device to be also
+used in other device types that are known to work with it.
+
+Image
+-----
+
+> An image represents a downloadable operating system image that can be booted
+  in a device or family of devices.
+
+_Some examples of images are: Raspbian Jessie, ResinOS, CoreOS, and Ostro._
+
+Here's an example of a real-world image manifest for Raspbian Jessie:
+
+```json
+{
+  "displayName": "Raspbian Jessie",
+  "version": "May 2016",
+  "url": "https://www.raspbian.org",
+  "releaseNotesUrl": "http://downloads.raspberrypi.org/raspbian/release_notes.txt",
+  "logoPath": "image-logo.svg",
+  "checksumType": "sha1",
+  "checksum": "64c7ed611929ea5178fbb69b5a5f29cc9cc7c157",
+  "recommendedDriveSize": 4294967296,
+  "bmapPath": "raspbian-jessie.bmap",
+  "instructionsPath": "raspbian-getting-started.markdown",
+  "updateUrl": "https://downloads.raspberrypi.org/raspbian_latest",
+  "etag": "c0170-53152af2-533d18ef29fc0",
+  "supportedDevices": [
+    "raspberrypi",
+    "raspberrypi2",
+    "raspberrypi3"
+  ]
+}
+```
+
+A image may be described with any of the following properties:
+
+### `displayName (String)`
+
+The human-friendly name of the image.
+
+### `version (String)`
+
+The version of the image.
+
+### `url (String)`
+
+The main url of the image.
+
+### `releaseNotesUrl (String)`
+
+The url to the image's release notes or CHANGELOG.
+
+### `logoPath (String)`
+
+The path to the image specific logo in SVG format.
+
+The SVG format requirement ensures that the logo looks good in all types of
+screen resolutions.
+
+### `checksumType (String)`
+
+The checksum type. The current possible values are: `sha1`, `crc32`, and `md5`.
+
+### `checksum (String)`
+
+The actual checksum according to the type declared in `checksumType`.
+
+### `bytesToZeroOutFromTheBeginning (Number)`
+
+The number of bytes to zero out from the beginning before starting the flash
+process. This option currently only applies when using bmaps.
+
+### `recommendedDriveSize (Number)`
+
+The recommended drive size to flash this image, in bytes.
+
+The use case for this option is that while a drive might be large enough to
+contain the image, it might not be large enough to deliver a good experience
+when actual using the application or operating system contained in the image.
+
+### `bmapPath (String)`
+
+Path to a [bmap][bmap] file.
+
+### `instructionsPath (String)`
+
+Path to a markdown file showing instructions to the user for when the flash
+completes.
+
+### `supportedDevices (String[])`
+
+An array of the device slugs this image supports.
+
+### `updateUrl (String)`
+
+The url where the client can request the latest version of the image.
+
+The way this works is that you provide a generic url that redirects to the
+latest image resource. The client will follow all redirects until it finds the
+final image, and will check if the resource matches the currently selected
+image.
```

Comment:
Yeah, there needs to be some more thinking through here. To be completely honest, the way we designed this was by looking at how the RPF does it now, so that it would be "automatically supported" by them, while being a fairly standard pattern that others can implement also. Happy to hear alternatives, and though not necessary, "RPF compatibility" is a definite nice-to-have